linkcheck/better_exchook2.py

```diff
@@ -122,14 +122,14 @@ def pp_extra_info(obj, depthlimit = 3):
 	s = []
 	if hasattr(obj, "__len__"):
 		try:
-			if type(obj) in (str,unicode,list,tuple,dict) and len(obj) <= 5:
+			if type(obj) in (str,list,tuple,dict) and len(obj) <= 5:
```

Comment:
Ah, but this raises a question: should we add bytes in here?

(What even is `better_exchook2.py`?  Is it the thing that produces internal error messages with those very verbose tracebacks?)

Reply:
> (What even is `better_exchook2.py`? Is it the thing that produces internal error messages with those very verbose tracebacks?)

Good point - yes it is only used in that one place.

Can we replace it? Is there anything now in the standard library that would be good enough? The one thing I do like is the clear printing of values.

Reply:
I don't think there's anything in the standard library.

Other traceback printing alternatives that I know of are

- Sentry, which captures the stack trace with locals and submits it to a webserver somewhere for aggregation, not suitable for linkchecker
- zope.exceptions has a custom traceback formatter that annotates each stack frame with the value of special `__traceback_info__` attribute, not a good choice for linkchecker
- pytest has a custom traceback formatter for test failures, I'm not sure if it's extractable

I haven't done any pypi or google searches for rich Python traceback libraries.

I think "don't fix what ain't broke" applies here and we can keep better_exchook2.  Having said that, I still think it may be valuable to restore special handling of bytestring values that as accidentally lost in the Python 3 transition.  In case bytestrings show up in locals, which has happened.  The thing is I'm not sure what the special handling _is_ because I haven't looked at the surrounding code.  Is it printing the length?  Truncating long values?  (We'd want that for bytestrings, I'm sure.)

Reply:
Thanks for describing those options. I think adding a dependency for this would need some careful thought. No problem adding bytes to the lists.